src/services/plugin.service.ts

```diff
@@ -12,6 +13,37 @@ export class PluginService {
     if (this.plugins[pluginName]) {
       throw new Error(`${pluginName} already exists as a plugin.`);
     }
-    return (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    const reference = (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    this.registerAliases(pluginName);
+
+    return reference;
+  }
+
+  registerAliases(pluginName: string, _args?: any): void {
+    const aliases = this.plugins[pluginName].pluginAlias;
+
+    if (aliases === undefined) {
+      this.aliases[pluginName] = pluginName;
+    } else {
+      if (!aliases.includes(pluginName)) {
+        if (this.aliases[pluginName] !== undefined) {
+          throw new Error(
+            `Duplicate alias detected: ${pluginName} is claiming its primary alias ${pluginName}, prevouisly claimed by ${this.aliases[pluginName]}.`
+          );
+        }
+        this.aliases[pluginName] = pluginName;
+      }
```

Comment:
So the main plugin name is also considered an alias?

Reply:
yes, unless we want to set plugin to.

`const plugin = plugins[command.name] || plugins[command.name];`